sdk/core/core/inc/az_span.h

```diff
@@ -20,93 +20,94 @@
 typedef struct {
   struct {
     uint8_t * begin;
-    size_t capacity;
-    size_t length;
+    int32_t length;
+    int32_t capacity;
   } _internal;
 } az_span;
 
-AZ_NODISCARD AZ_INLINE az_span az_span_empty() { return (az_span){ 0 }; }
+/********************************  SPAN GETTERS / SETTERS */
 
-AZ_NODISCARD AZ_INLINE bool az_span_is_empty(az_span const span) {
-  return span._internal.length == 0;
+/**
+ * @brief convenient way to get the capacity from one span
+ *
+ */
+AZ_NODISCARD AZ_INLINE int32_t az_span_capacity_get(az_span const span) {
+  return span._internal.capacity;
 }
 
 /**
- * @brief returns a span with the left @var n bytes of the given @var span.
+ * @brief convenient way to set the capacity from one span
  *
- * If the @var n is greater than the span capacity then the whole @var span is returned.
  */
-AZ_NODISCARD AZ_INLINE az_span az_span_take(az_span const span, size_t const n) {
-  if (span._internal.capacity <= n) {
-    return span;
-  }
-  return (az_span){ ._internal = { .begin = span._internal.begin, .capacity = n, .length = n } };
+AZ_NODISCARD AZ_INLINE void az_span_capacity_set(az_span span, int32_t const capacity) {
```

Comment:
This is not going to work, you need to take `az_span * const span`, and then set `span->_internal.capacity = capacity`. Otherwise, you're modifying a local copy.